tests/test_spsa_lmp_params.py

```
@@ -0,0 +1,59 @@
+
+
+def test_stale_array_mapping_raises_rather_than_silently_doing_nothing():
+    """The defect this whole change exists to prevent."""
+    build = _load_build()
+    constants = (REPO.parent / "rusty-rival/src/engine_constants.rs").read_text()
+    broken = constants.replace("[u8; 9]", "[u8; 7]")  # simulate the array growing again
+
+    try:
+        build.apply_parameters(broken, {"lmp_threshold_depth1": 5})
+    except ValueError as e:
+        assert "did not match" in str(e)
```

Ruff (0.16.1)

[warning] Found assertion on exception `e` in `except` block, use `pytest.raises()` instead

(PT017)